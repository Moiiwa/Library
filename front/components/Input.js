import React from 'react'
import { getBookWithIsbn } from '../api/isbn'
import { checkIsbnFormat } from './utils'

class Input extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            isbn: '',
            formatMessage: false
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({ isbn: event.target.value });
    }

    handleSubmit = async () => {
        if (checkIsbnFormat(this.state.isbn)) {
            this.setState({ formatMessage: false });
            const data = await getBookWithIsbn(this.state.isbn);
            console.log(data);
        } else {
            this.setState({ formatMessage: true });
        }
    }

    render() {
        return (
            <div>
                {this.state.formatMessage ?
                    <div style={{ color: 'red' }}>Isbn should be in 111-1-1111-1111-1 format</div> :
                    null
                }
                <input
                    type='text'
                    placeholder='111-1-1111-1111-1'
                    required={true}
                    value={this.state.isbn}
                    onChange={this.handleChange}
                />
                <button onClick={this.handleSubmit}>Find book</button>
            </div>
        );
    }
}

export default Input;