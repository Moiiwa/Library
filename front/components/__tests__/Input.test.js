import React from 'react';
import Input from '../Input';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Input', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<Input />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('updates input value', () => {
        const wrapper = shallow(<Input />);
        wrapper.find('input').simulate('change', { target: { value: '666-6-6666-6666-6' } });
        expect(wrapper.find('input').props().value).toEqual('666-6-6666-6666-6');
    });

    it('Updates isbn state', () => {
        const wrapper = shallow(<Input />);
        wrapper.find('input').simulate('change', { target: { value: '666-6-6666-6666-6' } });
        expect(wrapper.instance().state.isbn).toBe('666-6-6666-6666-6');
    });

    it('formatMessage state is true when right isbn', () => {
        const wrapper = shallow(<Input />);
        wrapper.find('input').simulate('change', { target: { value: '666-6-6666-6666-6' } });
        wrapper.find('button').simulate('click')
        expect(wrapper.instance().state.formatMessage).toBe(false);
    });

    it('formatMessage state is true when wrong isbn', () => {
        const wrapper = shallow(<Input />);
        wrapper.find('input').simulate('change', { target: { value: '666-6' } });
        wrapper.find('button').simulate('click')
        expect(wrapper.instance().state.formatMessage).toBe(true);
    });
});