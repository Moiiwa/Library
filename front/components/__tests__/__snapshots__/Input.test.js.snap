// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="111-1-1111-1111-1"
    required={true}
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Find book
  </button>
</div>
`;
